### PR TITLE
chartsvc: Unit and integration tests

### LINKED ISSUE #301: Chartsvc unit and integration tests
Unit tests for handler functions (similar to [ratesvc](https://github.com/kubeapps/ratesvc)), and integration tests similar to [Monocular](https://github.com/kubernetes-helm/monocular/blob/master/src/api/main_test.go).

### PR DESCRIPTION
**Pull request by sameersbn**
Closes #301

### CODE REVIEW

File: cmd/chartsvc/handler.go [hunk 1]
```diff
@@ -95,39 +95,39 @@ func listRepoCharts(w http.ResponseWriter, req *http.Request, params Params) {
 func getChart(w http.ResponseWriter, req *http.Request, params Params) {
 	db, closer := dbSession.DB()
 	defer closer()
-	var chart *models.Chart
```
Comment (prydonius) [inline cmd/chartsvc/handler.go]: what's the reasoning behind these pointer to value changes?

File: cmd/chartsvc/handler_test.go [hunk 2]
```diff
@@ -0,0 +1,666 @@
...
+			ID: "my-repo/my-chart", ChartVersions: []models.ChartVersion{{Version: "0.1.2"}, {Version: "0.1.0"}},
+		}},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cResponse := newChartResponse(&tt.chart)
+			assert.Equal(t, cResponse.Type, "chart", "response type is chart")
+			assert.Equal(t, cResponse.ID, tt.chart.ID, "chart ID should be the same")
+			assert.Equal(t, cResponse.Attributes.(models.Chart).ChartVersions, tt.chart.ChartVersions, "chart version in the response should be the same")
+		})
```
Comment (prydonius) [inline cmd/chartsvc/handler_test.go]: can we check that the latestChartVersion relationship points to tt.chart.ChartVersions[0]?

File: cmd/chartsvc/handler_test.go [hunk 3]
```diff
@@ -0,0 +1,666 @@
...
+		}},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			for i := range tt.chart.ChartVersions {
+				cvResponse := newChartVersionResponse(&tt.chart, tt.chart.ChartVersions[i])
+				assert.Equal(t, cvResponse.Type, "chartVersion", "response type is chartVersion")
+				assert.Equal(t, cvResponse.ID, tt.chart.ID+"-"+tt.chart.ChartVersions[i].Version, "reponse id should have chart version suffix")
+				assert.Equal(t, cvResponse.Attributes.(models.ChartVersion).Version, tt.chart.ChartVersions[i].Version, "chart version in the response should be the same")
```
Comment (prydonius) [inline cmd/chartsvc/handler_test.go]: check the selfLink and "chart" relationship here

File: cmd/chartsvc/handler_test.go (hunk 2 above)
Comment (prydonius) [reply cmd/chartsvc/handler_test.go]: also the selfLink

File: cmd/chartsvc/handler_test.go [hunk 4]
```diff
@@ -0,0 +1,666 @@
...
+		})
+	}
+}
+
+func Test_newChartVersionListResponse(t *testing.T) {
+	tests := []struct {
+		name  string
+		chart models.Chart
+	}{
+		{"my-chart", models.Chart{
```
Comment (prydonius) [inline cmd/chartsvc/handler_test.go]: add a test for no chart versions

File: cmd/chartsvc/handler.go (hunk 1 above)
Comment (sameersbn) [reply cmd/chartsvc/handler.go]: I was having trouble getting the mock interface match the expectation description. Simplest was to fix this was to remove the double reference.

```console
mock: Unexpected Method Call
-----------------------------

One(**models.Chart)
		0: (**models.Chart)(0xc4200b41a8)

The closest call I have is: 

One(*models.Chart)
		0: &models.Chart{ID:"", Name:"", Repo:models.Repo{Name:"", URL:""}, Description:"", Home:"", Keywords:[]string(nil), Maintainers:[]models.Maintainer(nil), Sources:[]string(nil), Icon:"", RawIcon:[]uint8(nil), ChartVersions:[]models.ChartVersion(nil)}
```

File: cmd/chartsvc/handler_test.go (hunk 2 above)
Comment (sameersbn) [reply cmd/chartsvc/handler_test.go]: done!

File: cmd/chartsvc/handler_test.go (hunk 3 above)
Comment (sameersbn) [reply cmd/chartsvc/handler_test.go]: done

File: cmd/chartsvc/handler_test.go (hunk 4 above)
Comment (sameersbn) [reply cmd/chartsvc/handler_test.go]: done!

**Comment by sameersbn** [comment]
updated the PR. Please review.

File: cmd/chartsvc/handler.go (hunk 1 above)
Comment (prydonius) [reply cmd/chartsvc/handler.go]: hmm okay, I think the tests in monocular had a similar thing though, but it's not a big problem either way.

**Comment by prydonius** [comment]
thanks for the separate commits, was a lot easier to review! lgtm!

**Pull request merged**